lib/Installation/ProductSelection/ProductSelectionPage.pm

```diff
@@ -74,4 +76,19 @@ sub select_SMGR_Retail {
     $self->{rb_SMGR_Retail}->select();
 }
 
+sub switch_keyboard_layout {
+    my ($self, $keyboard_layout) = @_;
+    $self->{cb_keyboard_layout}->select($keyboard_layout);
+}
+
+sub enter_keyboard_test {
+    my ($self, $test) = @_;
+    $self->{tb_keyboard_test}->set($test);
```

Comment:
This seems  hacky. Since we cannot actually send the $test value to the test field, maybe we could use another action, instead of set and not send any value at all? Or there might be another action that can actually send text? What have you tried so far?

Reply:
That is the only action in server side that set the focus for now afik, I'm not sure if we could implement that in server side in a way that we could call it set_focus (if I'm getting right your comment) we would need to check that first. For simplicity I keep the same name that we use for other textboxes, because call it set_focus when we really set_text is deceiving.